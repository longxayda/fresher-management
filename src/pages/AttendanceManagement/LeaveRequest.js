import React, { useEffect, useState, useMemo } from "react";
import {
  useTable,
  usePagination,
  useGlobalFilter,
  useSortBy,
  useFilters,
} from "react-table";
import { addRequest as addRequestAction } from "redux/attendanceManagementSlice/leaveRequestSlice";
import { useRowSelect } from 'react-table'
import { Table, Input, Label, Badge } from "reactstrap";
import { Button, Col, Row, Container } from "react-bootstrap";
import { Form } from "react-bootstrap";
import "assets/css/attendance-table.css"
import "@fortawesome/fontawesome-free/css/all.min.css";
import RequestModal from "components/LeaveRequest/Modal/modalRequest";
import ModalDelete from "components/LeaveRequest/Modal/modalDelete";
import NotificationAlert from "react-notification-alert";
import notify from "components/LeaveRequest/Alert/alert";
import { useDispatch, useSelector } from "react-redux";
import { fetchLeaveRequest } from "redux/attendanceManagementSlice/leaveRequestSlice";
import { leaveRequestSelector, addRequestSelector } from "redux/selectors/attendanceSelectors/traineeAttendanceSelectors";


function DefaultColumnFilter({
  column: { filterValue, preFilteredRows, setFilter },
}) {
  return (
    <Form.Group className="mb-0"/>
  );
}

function fuzzyTextFilterFn(rows, id, filterValue) {
  return matchSorter(rows, filterValue, { keys: [(row) => row.values[id]] });
}
fuzzyTextFilterFn.autoRemove = (val) => !val;

function filterGreaterThan(rows, id, filterValue) {
  return rows.filter((row) => {
    const rowValue = row.values[id];
    return rowValue >= filterValue;
  });
}
filterGreaterThan.autoRemove = (val) => typeof val !== "number";

export function LeaveRequest() {
  const data = useSelector(leaveRequestSelector);
  if (data == null || typeof(data) != "object"  ) return <Container></Container>

  const dispatch = useDispatch();

  const notificationAlertRef = React.useRef(null);

  const [requestArr, setRequestArr] = useState([]);

  useEffect(() => {
    const traineeId = 3
    dispatch(
      fetchLeaveRequest({
        traineeId
      })
    );
  }, [dispatch]);

  //Add new request
  const [showAddRequest, setShowAddRequest] = useState(false);
  const [newRequest, setNewRequest] = useState({
    traineeID: 2,
    requestType: "Vắng",
    partial: "",
    startDate: "",
    endDate: "",
    reason: ""
  });
  const handleShowAddRequest = () => {
    setRequestArr([]);
    setShowAddRequest(true);
  };
  const handleCloseAddRequest = () => {
    setNewRequest({
      traineeID: 2,
      requestType: "Vắng",
      partial: "",
      startDate: "",
      endDate: "",
      reason: ""
     });
    setShowAddRequest(false);
  };
  const completeAddRequest = (newRequest) => {
    dispatch(
      addRequestAction({
        partial: newRequest.partial,
        startDate: newRequest.startDate,
        endDate: newRequest.endDate,
        reason: newRequest.reason,
      })
    );
    notify(
      "tr",
      "success",
      "Add new role successfully !!!",
      notificationAlertRef
    );
  };
  const handleAddRequest = () => {
    const newRequest = requestArr.map((request) => {
      return request.value;
    });
    completeAddRequest({
      id: Math.floor(Math.random() * 100) + 1,
      trainee_id: 3,
      requestType: "Vắng",
      startDate: newRequest.startDate,
      endDate: newRequest.endDate,
      partial: newRequest.partialType,
      detailReason: newRequest.reason,
      status: "Pending"
    });
    handleCloseAddRole();
  };



  function handleContent({ row }) {
    let startDate = data[row.index].startDate;
    let endDate = data[row.index].endDate;
    let content = "Request for absence from " + String(startDate) + " to " + String(endDate);
    return <a href="#" style={{ color: "blue", textDecoration: "none" }}>{content}</a>
  }

  function handleStatus({ value }) {
    let className = "";
    switch (value) {
      case "Pending":
        className = "bg-warning";
        break;

      case "Cancelled":
        className = "bg-secondary";
        break;

      case "Denied":
        className = "bg-danger";
        break;

      case "Approved":
        className = "bg-success";
        break;
    }
    return (
      <>
        <Badge className={className} style={{ fontSize: "1rem" }}>
          {value}
        </Badge>
      </>
    );
  }

  function handleDelete({ row }) {
    if (data[row.index].status == "Pending") {
      return <ModalDelete />
    }
    return <div />;
  }

  const columns = useMemo(
    () => data[0]
    ?
    [
      {
                Header: "#",
                accessor: "id",
              },
              {
                Header: "Content",
                accessor: "Some content",
                Cell: handleContent,
              },
              {
                Header: "Detailed Reason",
                accessor: "detailReason",
              },
              {
                Header: "Status",
                accessor: "status",
                Cell: handleStatus,
              },
              {
                Header: "Start Date",
                accessor: "startDate",
              },
              {
                Header: "End Date",
                accessor: "endDate",
              },
              {
                Header: "Partial",
                accessor: "partial",
              },
              {
                Header: "Action",
                accessor: "Some status",
                Cell: handleDelete,
              },
    ]
    :
    [
      {
        Header: "Data is loading, please wait...",
        columns: [
          {
            Header: "",
            accessor: "Data is loading, please wait..."
          },
        ],
      }
    ],
    [data]
  );

  const defaultColumn = React.useMemo(
    () => ({
      Filter: DefaultColumnFilter,
    }),
    []
  );

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    setGlobalFilter,
    rows,
    prepareRow,
    page,
    canPreviousPage,
    canNextPage,
    pageOptions,
    pageCount,
    gotoPage,
    nextPage,
    previousPage,
    setPageSize,
    state: { pageIndex, pageSize, globalFilter },
  } = useTable({ columns, data, defaultColumn },
    useFilters,
    useGlobalFilter,
    useSortBy,
    usePagination,
    useRowSelect,
  )

  React.useEffect(() => {}, [globalFilter]);

  return (
    <>
      <div className='wrap-table-content'>
        <Container>
        <div className="d-flex align-items-center justify-content-between pb-3">
            <div className="d-flex align-items-center justify-content-start">
              <div>
                <RequestModal
                newRequest={newRequest}
                requestArr={requestArr}
                showAddRequest={showAddRequest}
                setNewRequest={setNewRequest}
                handleCloseAddRequest={handleCloseAddRequest}
                setRequestArr={setRequestArr}
                handleAddRequest={handleAddRequest} />
              </div>
            </div>
            <div className="d-flex align-items-center justify-content-end">
              <div className="me-3">
                <Label className="mb-0">Search: </Label>
              </div>
              <div className=" mb-1">
                <Input
                  type="search"
                  value={globalFilter || ""}
                  onChange={(e) => setGlobalFilter(e.target.value)}
                />
              </div>
            </div>
          </div>

            <Row style={{ overflow: "auto" }} className="pt-2">
              <Container>
                <pre className="chorma">
                  <Table striped hover {...getTableProps()}>
                    <thead>
                      {headerGroups.map((headerGroup) => (
                        <tr {...headerGroup.getHeaderGroupProps()}>
                          {headerGroup.headers.map((column) => (
                            <th {...column.getHeaderProps()}>
                              <div
                                className="heading"
                                style={{
                                  fontSize: "14px",
                                  textTransform:"uppercase",
                                  color:"black",
                                  fontWeight: "bold",
                                  borderTop:"none!important",
                                  borderBottom:"none",
                                  textAlign: "center",
                              }}
                            >
                              {column.render("Header")}
                            </div>
                          </th>
                        ))}
                      </tr>
                    ))}
                  </thead>
                  <tbody {...getTableBodyProps()}>
                    {page.map((row, i) => {
                      prepareRow(row);
                      return (
                        <tr {...row.getRowProps()}>
                          {row.cells.map((cell) => {
                            return (
                              <td {...cell.getCellProps()}>{cell.render("Cell")}</td>
                            );
                          })}
                        </tr>
                      );
                    })}
                  </tbody>
                  </Table>
                </pre>
              </Container>
            </Row>
            <Row>
            {data ? <Col className="pagination d-flex align-items-center justify-content-center">
              <Button
                className="btn-fill btn-sm me-3"
                onClick={() => gotoPage(0)}
                disabled={!canPreviousPage}
              >
                {"<<"}
              </Button>
              <Button
                className="btn-fill btn-sm me-3"
                onClick={() => previousPage()}
                disabled={!canPreviousPage}
              >
                {"<"}
              </Button>
              <Button
                className="btn-fill btn-sm me-3"
                onClick={() => nextPage()}
                disabled={!canNextPage}
              >
                {">"}
              </Button>
              <Button
                className="btn-fill btn-sm me-5"
                onClick={() => gotoPage(pageCount - 1)}
                disabled={!canNextPage}
              >
                {">>"}
              </Button>
              <span className="me-3 ml-5">
                Page
                <strong>
                  {' '}{pageIndex + 1} of {pageOptions.length}
                </strong>
              </span>
              <span className="me-3"> Go to page: </span>
              <Input
                type="number"
                className="me-3"
                defaultValue={pageIndex + 1}
                onChange={(e) => {
                  const page = e.target.value ? Number(e.target.value) - 1 : 0;
                  gotoPage(page);
                }}
                style={{ width: "100px" }}
              />
              <div className="">
                <Input
                  type="select"
                  value={pageSize}
                  onChange={(e) => {
                    setPageSize(Number(e.target.value));
                  }}
                >
                  {[10, 20, 30, 40, 50].map((pageSize) => (
                    <option key={pageSize} value={pageSize}>
                      Show {pageSize}
                    </option>
                  ))}
                </Input>
              </div>
              <br />
            </Col> : <Col></Col>}
          </Row>
        </Container>
      </div>
    </>
  );

}

export default LeaveRequest;
