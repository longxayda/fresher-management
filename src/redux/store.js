import {configureStore} from "@reduxjs/toolkit";
import attendTableOneTraineeSlice from "./attendanceManagementSlice/attendTableOneTraineeSlice";

// import ticketClassSlice from "./attendanceManagementSlice/ticketClassSlice";
import diligentSummarySlice from "./attendanceManagementSlice/diligentSummarySlice";
import leaveRequestSlice from "./attendanceManagementSlice/leaveRequestSlice";
import checkboxFRReducer from "./fresherManageSlice/checkboxFresherSlice";
import evaluationSlice from "./auditorsSlice/evaluationSlice";
import resourceSlice from "./auditorsSlice/resourceSlice";
import auditorSlice from "./auditorsSlice/auditorSlice";
import manageFRReducer from "./fresherManageSlice/fresherManagementSlice";
import bonusPenaltyReducer from "./fresherManageSlice/penaltySlice";

import classSlice from "./classManagementSlice/classSlice";
import classDetailsSlice from "./classManagementSlice/classDetailsSlice";
import learningPathSlice from "./classManagementSlice/learningPathSlice";
import syllabusSlice from "./classManagementSlice/syllabusSlice";
import moduleSlice from "./classManagementSlice/moduleSlice";
import topicSlice from "./classManagementSlice/topicSlice";

import scoreSlice from "./fresherManageSlice/scoreSlice";
import topicMarkSlice from "./fresherManageSlice/topicMarkSlice";
import scheduleSlice from "./auditorsSlice/scheduleSlice";
import createNewAuditSlice from "./auditorsSlice/createNewAuditSlice";
import adminManageAttendanceReducer from "./attendanceManagementSlice/adminManageAttendanceSlice";
import adminTicketReducer from "./attendanceManagementSlice/adminTicketSlice";

import traineeAttendanceTableSlice from "./attendanceManagementSlice/traineeAttendanceTableSlice";
import authSlice from "./authSlice/authSlice";
import roleSlice from "./roleSlice";
import UserListSlice from "./UserListSlices/UserListSlice";
import attendanceSlice from "./attendanceManagementSlice/attendanceSlice";
import traineeUserProfileSlice from "./userProfileSlice/traineeUserProfileSlice";
import adminUserProfileSlice from "./userProfileSlice/adminUserProfileSlice";

import AddNewReportTableSlice from "components/WeeklyReportTables/addNewReportTableSlice";
import DetailReportTableSlice from "components/WeeklyReportTables/detailReportTableSlice";
import ReportListTableSlice from "components/WeeklyReportTables/reportListTableSlice";
import searchSlice from "components/Search/searchSlice";
import ReportModalSlice from "components/ReportModal/reportModalSlice";

import HistoryReportListTableSlice from "components/MonthlyReportTables/historyReportListTableSlice";
import coursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/coursesReportSlice";
import financeReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/financeReportSlice";
import detailCoursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/detailCourseReportSlice";
import detailFinanceReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/detailFinanceReportSlice";

import DetailDeliveryTableSlice from "components/MonthlyReportTables/detailDeliveryTableSlice";
import DeliveryReportListTableSlice from "components/MonthlyReportTables/deliveryReportListTableSlice";
import NewDeliveryTableSlice from "components/MonthlyReportTables/newDeliveryTableSlice";

import OutputsReportTableSlice from "components/OutputsReportTables/outputsReportTableSlice";
import fresherReportSlice from "../pages/OutputsReportPage/fresher/fresherReportSlice";
import svttReportSlice from "../pages/OutputsReportPage/svtt/svttReportSlice";

import detailOutputsReportSlice from "../pages/OutputsReportPage/detailOutputsReportSlice";
import detailFresherReportSlice from "../pages/OutputsReportPage/fresher/detailFresherReportSlice";
import detailSVTTReportSlice from "../pages/OutputsReportPage/svtt/detailSVTTReportSlice";

import thunk from "redux-thunk";
const store = configureStore({
	reducer: {
		Filters: searchSlice,
		evaluation: evaluationSlice,
		resource: resourceSlice,
		manageFR: manageFRReducer,
		checkboxFR: checkboxFRReducer,
		attendTableOneTrainee: attendTableOneTraineeSlice,
		// traineeTabTable: TraineeTabTableSlice,
		class: classSlice,
		learningPath: learningPathSlice,
		syllabus: syllabusSlice,
		classDetails: classDetailsSlice,
		module: moduleSlice,
		topicModule: topicSlice,
		auth: authSlice,
		role: roleSlice,
		// UserList
		UserList: UserListSlice,
		schedule: scheduleSlice,
		newAudit: createNewAuditSlice,
		auditor: auditorSlice,

		// attendTable: attendTableSlice,
		// ticketClass: ticketClassSlice,
		diligentSummary: diligentSummarySlice,
		leaveRequest: leaveRequestSlice,
		adminManageAttendance: adminManageAttendanceReducer,
		adminTicket: adminTicketReducer,
		attendance: attendanceSlice,
		traineeAttendanceTable: traineeAttendanceTableSlice,
		bonusPenalty: bonusPenaltyReducer,
		score: scoreSlice,
		topic: topicMarkSlice,
		traineeUserProfile: traineeUserProfileSlice,
		adminUserProfile: adminUserProfileSlice,

		ReportList: ReportListTableSlice,
		AddNewReport: AddNewReportTableSlice,
		DetailReport: DetailReportTableSlice,
		Filters: searchSlice,
		ReportModal: ReportModalSlice,

		HistoryReportList: HistoryReportListTableSlice,
		CoursesReport: coursesReportSlice,
		FinanceReport: financeReportSlice,
		DetailCoursesReport: detailCoursesReportSlice,
		DetailFinanceReport: detailFinanceReportSlice,
		DetailDeliveryReport: DetailDeliveryTableSlice,
		DeliveryReportList: DeliveryReportListTableSlice,
		NewDeliveryReport: NewDeliveryTableSlice,

		OutputsReport: OutputsReportTableSlice,
		FresherReport: fresherReportSlice,
		SVTTReport: svttReportSlice,
		DetailFresherReport: detailFresherReportSlice,
		DetailSVTTReport: detailSVTTReportSlice,
		DetailOutputsReport: detailOutputsReportSlice,
	},
	middleware: [thunk],
});

export default store;
