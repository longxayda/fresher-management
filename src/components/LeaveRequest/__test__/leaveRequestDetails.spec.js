import React from 'react';
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, mount } from 'enzyme';
import { WrapTable } from 'components/LeaveRequest/index';
import store from 'redux/store';
import { Provider } from "react-redux";
import LeaveRequest from 'components/LeaveRequest';

Enzyme.configure({ adapter: new Adapter() });

describe('Test Leave Request Details Wrap Table', () => {
    let component;

    test('render Leave Request Details Wrap Table\'s component with EMPTY props', () => {
        const column = [], data = [];
        component = shallow(<WrapTable columns = { column } data = { data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(true);

    });

    test('render Leave Request Details Wrap Table\'s component with NULL props', () => {
        const columns = null, data = null;
        component = shallow(<WrapTable columns = { columns } data = { data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(false);
    });

    test('render Leave Request Details Wrap Table\'s component with VALID props', () => {
        const columns = [
            {
                Header: "#",
                accessor: "id",
              },
              {
                Header: "Content",
                accessor: "content",
              },
              {
                Header: "Status",
                accessor: "status",
              },
              {
                Header: "Start Date",
                accessor: "startDate",
              },
              {
                Header: "End Date",
                accessor: "endDate",
              },
              {
                Header: "Action",
                accessor: "ticketStatus",
              }
        ];
        const data = [
            {
                id: "1",
                content: "01/07/2022",
                status: "Pending",
                startDate: "01/07/2022",
                endDate: '03/07/2022',
                ticketStatus: "Undone",
              }
        ];
        component = shallow(<WrapTable columns = { columns } data = { data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(true);

    });

    test('render Leave Request Details Wrap Table\'s component with INVALID props', () => {
        const columns = 1;
        const data = 'temp';
        component = shallow(<WrapTable columns = { columns } data = {data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(false);
    });

    test('render Leave Request Details Wrap Table\'s component with INVALID COLUMNS props', () => {
        const columns = 1;
        const data = [
            {
                id: "1",
                content: "01/07/2022",
                status: "Pending",
                startDate: "01/07/2022",
                endDate: '03/07/2022',
                ticketStatus: "Undone",
              }
          ];
        component = shallow(<WrapTable columns = { columns } data = { data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(false);
    });

    test('render Leave Request Details Wrap Table\'s component with INVALID DATA props', () => {
        const columns = [
            {
                Header: "#",
                accessor: "id",
              },
              {
                Header: "Content",
                accessor: "content",
              },
              {
                Header: "Status",
                accessor: "status",
              },
              {
                Header: "Start Date",
                accessor: "startDate",
              },
              {
                Header: "End Date",
                accessor: "endDate",
              },
              {
                Header: "Action",
                accessor: "ticketStatus",
              }
        ];
        const data = 'temp';
        component = shallow(<WrapTable columns = { columns } data = { data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(false);
    });

    test('render Leave Request Details Wrap Table\'s component with INVALID DATA TYPE', () => {
        const columns = 'temp columns';
        const data = 'temp';
        component = shallow(<WrapTable columns = { columns } data = { data } />);
        expect(component.find('.wrap-table-content').exists()).toEqual(false);
    });
})

describe('Test Leave Request Details Page', () => {
  let component;
  beforeEach(() => {
      component = mount(<Provider store = { store }> <LeaveRequest /> </Provider>);
  });

  test('has the correct detail', () => {
      expect(component.find('.leave-request-detail').exists()).toEqual(true);
  });

  test('render Leave Request Details Page\'s component correctly', () => {
      component = mount(<Provider store = { store }> <LeaveRequest /> </Provider>);
      expect(component.html()).toMatchSnapshot();
  });
})