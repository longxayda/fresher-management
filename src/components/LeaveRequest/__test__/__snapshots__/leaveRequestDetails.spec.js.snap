// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Leave Request Details Page render Leave Request Details Page's component correctly 1`] = `
" <div class=\\"leave-request-detail\\"><div class=\\"wrap-table-content container\\"><div class=\\"d-flex align-items-center justify-content-between pb-3\\"><div class=\\"me-3\\"><div class=\\"leave-request-modal\\"><button type=\\"button\\" class=\\"btn-fill btn btn-primary\\">Make leave request</button></div></div><div class=\\"d-flex align-items-center justify-content-between\\"><div class=\\"me-3\\"><label class=\\"mb-0\\">Search: </label></div><input type=\\"search\\" class=\\"form-control\\" value=\\"\\"></div></div><style>
td,tr,th{
text-align: center;
}
</style><pre class=\\"chorma\\"><table role=\\"table\\" class=\\"table table-striped table-hover\\"><thead><tr role=\\"row\\"><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\">#<div></div></div></th><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\">Content<div></div></div></th><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\">Status<div></div></div></th><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\">Start Date<div></div></div></th><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\">End Date<div></div></div></th><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\">Action<div></div></div></th></tr></thead><tbody role=\\"rowgroup\\"><tr role=\\"row\\"><td role=\\"cell\\">1</td><td role=\\"cell\\"><div class=\\"leave-request-handle-button\\"><div class=\\"text-primary\\" style=\\"cursor: pointer;\\">Request for absence from 01/07/2022 to 03/07/2022</div></div></td><td role=\\"cell\\"><span style=\\"font-size: 1rem;\\" class=\\"bg-warning badge badge-secondary\\">Pending</span></td><td role=\\"cell\\">01/07/2022</td><td role=\\"cell\\">03/07/2022</td><td role=\\"cell\\"><div><button type=\\"button\\" class=\\"ms-2 btn btn-danger\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"xmark\\" class=\\"svg-inline--fa fa-xmark \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 320 512\\"><path fill=\\"currentColor\\" d=\\"M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z\\"></path></svg></button></div></td></tr><tr role=\\"row\\"><td role=\\"cell\\">2</td><td role=\\"cell\\"><div class=\\"leave-request-handle-button\\"><div class=\\"text-primary\\" style=\\"cursor: pointer;\\">Request for absence from 05/07/2022 to 05/07/2022</div></div></td><td role=\\"cell\\"><span style=\\"font-size: 1rem;\\" class=\\"bg-secondary badge badge-secondary\\">Cancelled</span></td><td role=\\"cell\\">05/07/2022</td><td role=\\"cell\\">05/07/2022</td><td role=\\"cell\\"><div></div></td></tr><tr role=\\"row\\"><td role=\\"cell\\">3</td><td role=\\"cell\\"><div class=\\"leave-request-handle-button\\"><div class=\\"text-primary\\" style=\\"cursor: pointer;\\">Request for absence from 10/07/2022 to 15/07/2022</div></div></td><td role=\\"cell\\"><span style=\\"font-size: 1rem;\\" class=\\"bg-danger badge badge-secondary\\">Rejected</span></td><td role=\\"cell\\">10/07/2022</td><td role=\\"cell\\">15/07/2022</td><td role=\\"cell\\"><div></div></td></tr><tr role=\\"row\\"><td role=\\"cell\\">4</td><td role=\\"cell\\"><div class=\\"leave-request-handle-button\\"><div class=\\"text-primary\\" style=\\"cursor: pointer;\\">Request for absence from 20/07/2022 to 21/07/2022</div></div></td><td role=\\"cell\\"><span style=\\"font-size: 1rem;\\" class=\\"bg-success badge badge-secondary\\">Approved</span></td><td role=\\"cell\\">20/07/2022</td><td role=\\"cell\\">21/07/2022</td><td role=\\"cell\\"><div></div></td></tr><tr role=\\"row\\"><td role=\\"cell\\">5</td><td role=\\"cell\\"><div class=\\"leave-request-handle-button\\"><div class=\\"text-primary\\" style=\\"cursor: pointer;\\">Request for absence from 25/07/2022 to 25/07/2022</div></div></td><td role=\\"cell\\"><span style=\\"font-size: 1rem;\\" class=\\"bg-secondary badge badge-secondary\\">Cancelled</span></td><td role=\\"cell\\">25/07/2022</td><td role=\\"cell\\">25/07/2022</td><td role=\\"cell\\"><div></div></td></tr><tr role=\\"row\\"><td role=\\"cell\\">6</td><td role=\\"cell\\"><div class=\\"leave-request-handle-button\\"><div class=\\"text-primary\\" style=\\"cursor: pointer;\\">Request for absence from 01/08/2022 to 02/08/2022</div></div></td><td role=\\"cell\\"><span style=\\"font-size: 1rem;\\" class=\\"bg-success badge badge-secondary\\">Approved</span></td><td role=\\"cell\\">01/08/2022</td><td role=\\"cell\\">02/08/2022</td><td role=\\"cell\\"><div></div></td></tr><tr role=\\"row\\"><td role=\\"cell\\">7</td><td role=\\"cell\\"><div class=\\"leave-request-handle-button\\"><div class=\\"text-primary\\" style=\\"cursor: pointer;\\">Request for absence from 05/08/2022 to 07/08/2022</div></div></td><td role=\\"cell\\"><span style=\\"font-size: 1rem;\\" class=\\"bg-warning badge badge-secondary\\">Pending</span></td><td role=\\"cell\\">05/08/2022</td><td role=\\"cell\\">07/08/2022</td><td role=\\"cell\\"><div><button type=\\"button\\" class=\\"ms-2 btn btn-danger\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"xmark\\" class=\\"svg-inline--fa fa-xmark \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 320 512\\"><path fill=\\"currentColor\\" d=\\"M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z\\"></path></svg></button></div></td></tr><tr role=\\"row\\"><td role=\\"cell\\">8</td><td role=\\"cell\\"><div class=\\"leave-request-handle-button\\"><div class=\\"text-primary\\" style=\\"cursor: pointer;\\">Request for absence from 12/08/2022 to 20/08/2022</div></div></td><td role=\\"cell\\"><span style=\\"font-size: 1rem;\\" class=\\"bg-secondary badge badge-secondary\\">Cancelled</span></td><td role=\\"cell\\">12/08/2022</td><td role=\\"cell\\">20/08/2022</td><td role=\\"cell\\"><div></div></td></tr></tbody></table></pre><div class=\\"pagination d-flex align-items-center justify-content-center\\"><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-3 btn btn-primary\\">&lt;&lt;</button><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-3 btn btn-primary\\">&lt;</button><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-3 btn btn-primary\\">&gt;</button><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-5 btn btn-primary\\">&gt;&gt;</button><span class=\\"me-3 ml-5\\">Page <strong>1 of 1</strong></span><span class=\\"me-3\\"> Go to page: </span><input style=\\"width: 100px;\\" type=\\"number\\" class=\\"me-3 form-control\\" value=\\"1\\"><div class=\\"\\"><select class=\\"form-control\\"><option value=\\"10\\">Show 10</option><option value=\\"20\\">Show 20</option><option value=\\"30\\">Show 30</option><option value=\\"40\\">Show 40</option><option value=\\"50\\">Show 50</option></select></div></div></div><br></div> "
`;
